src/RequestHandlerSwooleRunner.php

```diff
@@ -81,43 +114,88 @@ function (Throwable $exception) use ($serverRequestErrorResponseGenerator) : Res
+        $this->docRoot = $config['options']['document_root'] ?? getcwd() . '/public';
     }
 
     /**
      * Run the application
      */
     public function run() : void
     {
-        $this->swooleHttpServer->on('start', function ($server) {
-            printf("Swoole is running at %s:%s\n", $server->host, $server->port);
-        });
-
-        $this->swooleHttpServer->on('request', function ($request, $response) {
-            printf(
-                "%s - %s - %s %s\n",
-                date('Y-m-d H:i:sO', time()),
-                $request->server['remote_addr'],
-                $request->server['request_method'],
-                $request->server['request_uri']
-            );
-            $emit = new SwooleEmitter($response);
-            try {
-                $psr7Request = ($this->serverRequestFactory)($request);
-            } catch (Throwable $e) {
-                // Error in generating the request
-                $this->emitMarshalServerRequestException($emit, $e);
-                return;
-            }
-            $emit->emit($this->handler->handle($psr7Request));
-        });
+        $this->swooleHttpServer->on('start', [$this, 'onStart']);
+        $this->swooleHttpServer->on('request', [$this, 'onRequest']);
         $this->swooleHttpServer->start();
     }
 
+    /**
+     * On start event for swoole http server
+     */
+    public function onStart(SwooleHttpServer $server) : void
+    {
+        printf("Swoole is running at %s:%s\n", $server->host, $server->port);
+    }
+
+    /**
+     * On HTTP request event for swoole http server
+     */
+    public function onRequest(
+        SwooleHttpRequest $request,
+        SwooleHttpResponse $response
+    ) {
+        printf(
+            "%s - %s - %s %s\n",
+            date('Y-m-d H:i:sO', time()),
+            $request->server['remote_addr'],
+            $request->server['request_method'],
+            $request->server['request_uri']
+        );
+        if ($this->getStaticResource($request, $response)) {
+            return;
+        }
+        $emit = new SwooleEmitter($response);
+        try {
+            $psr7Request = ($this->serverRequestFactory)($request);
+        } catch (Throwable $e) {
+            // Error in generating the request
+            $this->emitMarshalServerRequestException($emit, $e);
+            return;
+        }
+        $emit->emit($this->handler->handle($psr7Request));
+    }
+
+    /**
+     * Emit marshal server request exception
+     */
     private function emitMarshalServerRequestException(
         EmitterInterface $emitter,
         Throwable $exception
     ) : void {
         $response = ($this->serverRequestErrorResponseGenerator)($exception);
         $emitter->emit($response);
     }
+
+    /**
+     * Get a static resource, if any, and set the swoole HTTP response
+     */
+    private function getStaticResource(
+        SwooleHttpRequest $request,
+        SwooleHttpResponse $response
+    ) : bool {
```

Comment:
PSR-12 breakage, here and across entire codebase.
```When you have a return type declaration present there MUST be one space after the colon followed by the type declaration. The colon and declaration MUST be on the same line as the argument list closing parentheses with no spaces between the two characters.```

Reply:
As Zend Framework coding standard we support PSR-2 using our tool [zendframework/zend-coding-standard](https://github.com/zendframework/zend-coding-standard). PSR-12 is not still accepted, actually is moved back to draft status. We'll support it when it will be ready.

Reply:
Additionally, that particular syntax is being debated by the PSR-12 working group, as a number of folks (us included) prefer the `) : Type` syntax. ;-)